server/controllers/finance/fiscal.js

```diff
@@ -590,23 +594,25 @@ function closing(req, res, next) {
         const credit = value >= 0 ? Math.abs(value) : 0;
 
         const resultParams = [
-          projectId,                      // project_id
-          fiscal.id,                      // fiscal_year_id
-          fiscal.number_of_months + 1,    // period_id
-          fiscal.end_date,                // date : the last date of the fiscal year
-          'Ecriture du result lors de la cloture',
+          db.bid(uuid.v4()),
           accountId,
-          debit, credit,                  // debit and credit
-          debit, credit,                  // debit_equiv and credit_equiv in enterprise currency
-          currencyId,                     // enterprise currency because data came from period total
-          userId,                         // user id
+          debit,
+          credit,
+          voucher.uuid,
+          voucherDocumentUuid,
         ];
 
-        transaction.addQuery(sqlInsertJournal, resultParams);
+        transaction.addQuery(sqlInsertVoucherItem, resultParams);
       }
 
-      // lock the fiscal year
-      transaction.addQuery('UPDATE fiscal_year SET locked = 1 WHERE id = ?;', [id]);
+      // update the period to the period N+1 (13)
+      const sqlUpdateJournalPeriod = 'UPDATE posting_journal SET period_id = ? WHERE reference_uuid = ?';
+
+      // post voucher and close the fiscal year
+      transaction
```

Comment:
Hmm... I had originally imagined this not writing into the Posting Journal at all to prevent accountants from deleting values or messing with numbers there.  Originally, the idea was to just calculate and write the difference to `period_total`s.  I believe @sfount understood that too.  See the relevant issue: https://github.com/IMA-WorldHealth/bhima-2.X/issues/1651

Is there an issue with that logic?  Has a discussion happened that hasn't been posted on Github?